web/controllers/task_controller.ex

```diff
@@ -100,23 +119,40 @@ defmodule CoursePlanner.TaskController do
     end
   end
 
-  @error_messages %{
-    not_found: "Task was not found.",
-    already_finished: "Cannott grab as task is already finished.",
-    already_assigned: "Cannot grab as task is already assigned."
-  }
-
-  def grab(%{assigns: %{current_user: %{id: user_id}}} = conn, %{"task_id" => task_id}) do
-    case Tasks.grab(task_id, user_id, Timex.now()) do
+  def grab(%{assigns: %{current_user: %{id: volunteer_id}}} = conn, %{"task_id" => task_id}) do
```

Comment:
for now I don't see any problem extracting the response formatting to another function